pallets/oracle/src/lib.rs

```diff
@@ -251,26 +220,29 @@ pub mod pallet {
 impl<T: Config> Pallet<T> {
 	// public only for testing purposes
 	pub fn begin_block(_height: T::BlockNumber) {
-		// read to a temporary value, because we can't alter the map while we iterate over it
-		let raw_values_updated: Vec<_> = RawValuesUpdated::<T>::iter().collect();
+		let oracle_keys: Vec<_> = OracleKeys::<T>::get();
 
 		let current_time = Self::get_current_time();
 
 		let mut updated_items = Vec::new();
-		for (key, is_updated) in raw_values_updated.iter() {
-			if *is_updated || Self::is_outdated(key, current_time) {
-				let new_value = Self::update_aggregate(key);
-				updated_items.push((key.clone(), new_value));
+		let max_delay = Self::get_max_delay();
+		for key in oracle_keys.iter() {
+			let price = Self::get_timestamped(key);
+			let Some(price) = price else{
+				continue;
+			};
+			let is_outdated = current_time > price.timestamp + max_delay;
+			if !is_outdated {
+				updated_items.push((key.clone(), Some(price.value)));
 			}
 		}
-
+		let updated_items_len = updated_items.len();
 		if !updated_items.is_empty() {
 			Self::deposit_event(Event::<T>::AggregateUpdated { values: updated_items });
 		}
```

Comment:
Doing it this way makes it different from the original implementation.
In the 'original' implementation, the entries of `updated_items` can be set to `None` or `Some(price)`. But since you exit the for loop in l. 232 if the price is empty, the `updated_items` cannot contain values of type `None`. 

I think we should change it so that a value of `None` is pushed to `updated_items` if no price was found.